harness/determined/cli/command.py

```diff
@@ -319,9 +359,14 @@ def parse_config_overrides(config: Dict[str, Any], overrides: Iterable[str]) ->
             if key in _CONFIG_PATHS_COERCE_TO_LIST:
                 value = [value]
 
-        # TODO(#2703): Consider using full JSONPath spec instead of dot
-        # notation.
-        config = _set_nested_config(config, key.split("."), value)
+        # Convert config override in dot notation to json path
+        config_arg_in_json = _dot_to_json(key, value)
+        # Convert NoneType value None in config to empty string {}
+        config = _fix_none_value_in_config(config, key.split("."))
+        # Merge two objects in json format
+        config = merge_dicts(config, config_arg_in_json)
```

Comment:
The problem is that merge_dicts creates a new dict to return -- so that causes you to need to return it.

Reply:
Yes, merge_dicts does not modify the passed-in config, instead, it created and worked on the deep copy of the config. So if the returned new dict is re-assigned to variable config, does that make the config the local scope? It seems that is the only explanation for why I have to make it a return value.

Reply:
Does "_fix_none_value_in_config()" need the return too?

Maybe changing "config = merge_dicts(...)" to "merged_config = merge_dicts(...)" might be clearer? Then you can return "merged_config"?

Reply:
https://net-informations.com/python/iq/arguments.htm
```
Is Python call-by-value or call-by-reference?

In terms of official [Python documentation](https://docs.python.org/2/faq/programming.html#how-do-i-write-a-function-with-output-parameters-call-by-reference) ,arguments are passed by assignment in Python. The actual parameters (arguments) to a function call are introduced in the local symbol table of the called function when it is called; thus, arguments are passed using call by value (where the value is always an object reference, not the value of the object). Thus, if you change the value of the parameter within a function, the change is reflected in the calling function .
```

You can change the dict since it is mutable, but cannot change which dict the caller is referencing as you are doing in this case.

Reply:
> Does "_fix_none_value_in_config()" need the return too?
> 
> Maybe changing "config = merge_dicts(...)" to "merged_config = merge_dicts(...)" might be clearer? Then you can return "merged_config"?

sure

Reply:
> > Does "_fix_none_value_in_config()" need the return too?
> > Maybe changing "config = merge_dicts(...)" to "merged_config = merge_dicts(...)" might be clearer? Then you can return "merged_config"?
> 
> sure

Actually, it might not help that much in readability, because of the multiple --config we have to handle in a for loop: 

for config_arg in overrides:
...
+        merged_config = merge_dicts(config, config_arg_in_json)
+        config = merged_config

+    return merged_config